modules/app/src/main/java/org/locationtech/jtstest/function/SpatialIndexFunctions.java

```diff
@@ -117,6 +117,14 @@ public static Geometry strTreeNN(Geometry geoms, Geometry geom)
     return (Geometry) result;
   }
 
+  public static Geometry[] strTreeNNk(Geometry geoms, Geometry geom)
+  {
+    STRtree index = buildSTRtree(geoms);
+    int k = 10;
+    Object[] result = index.nearestNeighbour(geom.getEnvelopeInternal(), geom, new GeometryItemDistance(), k);
+    return (Geometry[]) result;
```

Comment:
Not sure TestBuilder handles Geometry[] results.  I'll have to check this, or perhaps you can run it and see.

It should be enhanced to automatically convert them to GeometryCollections, but that's not in place yet.